homeassistant/components/deconz/light.py

```diff
@@ -208,6 +223,15 @@ def __init__(self, device: _LightDeviceT, hub: DeconzHub) -> None:
         if device.effect is not None:
             self._attr_supported_features |= LightEntityFeature.EFFECT
             self._attr_effect_list = [EFFECT_COLORLOOP]
+
+            # For lights that report supported effects.
+            if isinstance(device, Light) and device.supported_effects is not None:
+                self._attr_effect_list = [
+                    EFFECT_TO_DECONZ[el]
+                    for el in device.supported_effects
+                    if el in EFFECT_TO_DECONZ
+                ]
+
             if device.model_id in ("HG06467", "TS0601"):
                 self._attr_effect_list = XMAS_LIGHT_EFFECTS
```

Comment:
As there is now logic to identify a light lets move everything in behind that check
```suggestion
            if isinstance(device, Light):
                if device.supported_effects is not None:
                    self._attr_effect_list = [
                        EFFECT_TO_DECONZ[el]
                        for el in device.supported_effects
                        if el in EFFECT_TO_DECONZ
                    ]

                if device.model_id in ("HG06467", "TS0601"):
                    self._attr_effect_list = XMAS_LIGHT_EFFECTS
```

Reply:
Done!